include/seqan3/alignment/cigar_conversion/alignment_from_cigar.hpp

```diff
@@ -0,0 +1,226 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2022, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2022, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the function seqan3::alignment_from_cigar.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <vector>
+
+#include <seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp>
+#include <seqan3/alignment/decorator/gap_decorator.hpp>
+#include <seqan3/alphabet/cigar/cigar.hpp>
+#include <seqan3/utility/views/slice.hpp>
+
+namespace seqan3
+{
+
+/*!\brief Construct an alignment from a cigar string and the corresponding sequnces.
+ * \ingroup cigar_conversion
+ * \tparam reference_type The type of the reference sequence for a SAM record.
+ * \tparam sequence_type The type of the read sequence for a SAM record.
+ * \param[in] cigar_vector The cigar information to convert to an alignment.
+ * \param[in] reference The reference sequence of the alignment represented by `cigar_vector`.
```

Comment:
What does "represented by cigar_vector" mean? Isn't just the query represented by it? As its the conversion from the ref to the read? I'm not sure if this is the best way to desribe this.

Reply:
I like both solutions. Maybe the first one a bit more.